src/HW6/sortList.c

```diff
@@ -0,0 +1,164 @@
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define GREEN(string) "\x1b[32m" string "\x1b[0m"
+#define RED(string) "\x1b[31m" string "\x1b[0m"
+
+typedef struct ListNode {
+    int value;
+    struct ListNode* prev;
+    struct ListNode* next;
+} ListNode;
+
+typedef struct List {
+    ListNode* head;
+    ListNode* tail;
+} List;
+
+List new()
+{
+    List list = { .head = NULL, .tail = NULL };
+    return list;
+}
+
+int push(List* list, int value)
+{
+    if (list == NULL)
+        return -1;
+
+    ListNode* new = malloc(sizeof(ListNode));
+    if (new == NULL)
+        return -2;
+
+    new->prev = new->next = NULL;
+    new->value = value;
+
+    if (list->head) {
+        list->head->prev = new;
+        new->next = list->head;
+        list->head = new;
+    } else {
+        list->head = list->tail = new;
+    }
+    ListNode* current = list->head;
+    while (current && current->next) {
+        if (current->value < current->next->value) {
+            int val = current->value;
+            current->value = current->next->value;
+            current->next->value = val;
+        }
+        current = current->next;
+    }
+    return 0;
+}
+
+void printList(List* list)
+{
+    ListNode* current = list->head;
+
+    while (current) {
+        printf("%d ", current->value);
+        current = current->next;
+    }
+}
+
+int delElement(List* list, int value)
```

Comment:
```suggestion
int deleteElement(List* list, int value)
```
Буквы вроде бесплатные